File: .ci/doc/java-project/src/main/java/SnippetTest.java
import io.kuzzle.sdk.Kuzzle;
import io.kuzzle.sdk.protocol.WebSocket;
import io.kuzzle.sdk.coreClasses.responses.Response;

import java.util.concurrent.ConcurrentHashMap;
import java.util.ArrayList;

public class SnippetTest {
  private static Kuzzle kuzzle;

  public static void main(String[] argv) {
    try {
      kuzzle = new Kuzzle(new WebSocket("kuzzle"));
      kuzzle.connect();
        ConcurrentHashMap<String, Object> document = new ConcurrentHashMap<>();
        document.put("key", "value");

        Boolean result = kuzzle
        .getDocumentController()
        .validate("nyc-open-data", "yellow-taxi", document)
        .get();
      System.out.println(result.toString());
    } catch (Exception e) {
      e.printStackTrace();
    } finally {
      if (kuzzle != null) {
        kuzzle.disconnect();
      }
    }
  }
}

File: .ci/doc/kotlin-project/src/main/java/SnippetTest.kt
import io.kuzzle.sdk.Kuzzle
import io.kuzzle.sdk.protocol.WebSocket
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.ExecutionException
import io.kuzzle.sdk.coreClasses.responses.Response

fun main() {
  val ws = WebSocket("kuzzle")
  val kuzzle = Kuzzle(ws).apply {
    connect()
  }
  try {
    val document: ConcurrentHashMap<String, Any?> =
      ConcurrentHashMap<String, Any?>().apply {
        put("key", "value")
      }


    val result: Boolean =
      kuzzle
      .documentController
      .validate("nyc-open-data", "yellow-taxi", document)
      .get()
  
    println(result.toString())
  } catch (e: Exception) {
    e.printStackTrace()
  } finally {
    kuzzle.disconnect()
  }
}
